lib/core/src/animation/animator/easing.rs

```diff
@@ -0,0 +1,92 @@
+//! This file provides the implementation of EasingAnimator.
+
+use super::ContinuousAnimator;
+use crate::animation::easing::FnEasing;
+use crate::control::event_loop::EventLoop;
+use crate::math::utils::linear_interpolation;
+use crate::math::utils::Interpolable;
+
+use nalgebra::clamp;
+use std::rc::Rc;
+use std::cell::RefCell;
+
+
+
+// ===============================
+// === EasingAnimationCallback ===
+// ===============================
+
+pub trait EasingAnimationCallback<T> = FnMut(T) + 'static;
+
+
+
+// ==========================
+// === EasingAnimatorData ===
+// ==========================
+
+struct EasingAnimatorData<T:Interpolable<T>+'static> {
+    initial_value       : T,
+    final_value         : T,
+    duration_seconds    : f64,
+    continuous_animator : Option<ContinuousAnimator>
+}
+
+
+
+// ======================
+// === EasingAnimator ===
+// ======================
+
+/// This struct animates from `origin_position` to `target_position` using easing functions.
+pub struct EasingAnimator<T:Interpolable<T>+'static> {
+    data : Rc<RefCell<EasingAnimatorData<T>>>
+}
+
+impl<T:Interpolable<T>+'static> EasingAnimator<T> {
+    /// Creates an EasingAnimator using a `easing_function` to move `object` from
+    /// `initial_value` to `final_value` in `duration_seconds`.
+    pub fn new<F,C>
+    ( mut event_loop                : &mut EventLoop
+    , mut easing_animation_callback : C
+    , easing_function               : F
+    , initial_value                 : T
+    , final_value                   : T
+    , duration_seconds              : f64) -> Self
+    where F:FnEasing, C:EasingAnimationCallback<T> {
```

Comment:
I highly appreciate how you make the code layout look nice, good job! ❤️

I was thinking about this example, and I believe that we could get a more clear layout this way:

```rust
    pub fn new<F:FnEasing, C:EasingAnimationCallback<T>>
    ( mut event_loop                : &mut EventLoop
    , mut easing_animation_callback : C
    , easing_function               : F
    , initial_value                 : T
    , final_value                   : T
    , duration_seconds              : f64) -> Self {
```

The reason for that is that the first line has already the variables and you left it empty already. So there is a place to define what `F` and what `C` is. I would not have so big problem if you used more descriptive names than `F` and `C`. It is not bad to use short names if they are described, and when I see `F:FnEasing` it is perfectly clear what happens here :)